test/test_kernels/test_aggressive_momentum_model_free_kernel.py

```diff
@@ -0,0 +1,331 @@
+    kernel.accept_matrix_updates()
+    np.testing.assert_array_equal(
+        kernel.previous_low_rank_candidate_L, low_rank_candidate_L
+    )
+    np.testing.assert_array_equal(kernel.previous_utility_matrix_Z, utility_matrix_Z)
+
+
+def test_aggressive_momentum_reject_matrix_updates(fixture_with_tol: Fixture) -> None:
+    (_, _, kernel) = fixture_with_tol
+
+    previous_utility_matrix_Z = kernel.previous_utility_matrix_Z.copy()
+    utility_matrix_Z = previous_utility_matrix_Z + 1.0
+
+    previous_low_rank_candidate_L = kernel.previous_low_rank_candidate_L.copy()
+    low_rank_candidate_L = previous_low_rank_candidate_L * 1.1
+
+    kernel.low_rank_candidate_L = low_rank_candidate_L
+    kernel.utility_matrix_Z = utility_matrix_Z
+
+    kernel.reject_matrix_updates()
+    np.testing.assert_array_equal(
+        kernel.low_rank_candidate_L, previous_low_rank_candidate_L
+    )
+    np.testing.assert_array_equal(kernel.utility_matrix_Z, previous_utility_matrix_Z)
+
+
+@patch(f"{PKG}.{KERNEL_CLASS}.reject_matrix_updates")
+@patch(f"{PKG}.{KERNEL_CLASS}.decrease_momentum_parameters")
+@patch(f"{PKG}.{KERNEL_CLASS}.accept_matrix_updates")
+@patch(f"{PKG}.{KERNEL_CLASS}.increase_momentum_parameters")
+@patch(f"{PKG}.{KERNEL_CLASS}.compute_parameter_update_loss")
+def test_aggressive_momentum_parameter_adaption(
+    mock_compute_parameter_update_loss: Mock,
+    mock_increase_momentum_parameters: Mock,
+    mock_accept_matrix_updates: Mock,
+    mock_decrease_momentum_parameters: Mock,
+    mock_reject_matrix_updates: Mock,
+    fixture_with_tol: Fixture,
+) -> None:
+    (_, _, kernel) = fixture_with_tol
+    kernel.elapsed_iterations = 3
+
+    # loss decreasing
+    kernel.parameter_update_loss = 5.0
+    mock_compute_parameter_update_loss.return_value = 4.0
+    kernel.step()
+
+    mock_compute_parameter_update_loss.assert_called_once()
+    mock_increase_momentum_parameters.assert_called_once()
+    mock_accept_matrix_updates.assert_called_once()
+    mock_decrease_momentum_parameters.assert_not_called()
+    mock_reject_matrix_updates.assert_not_called()
+
+    mock_compute_parameter_update_loss.reset_mock()
+    mock_increase_momentum_parameters.reset_mock()
+    mock_accept_matrix_updates.reset_mock()
+
+    # loss increasing
+    mock_compute_parameter_update_loss.return_value = 5.0
+    kernel.step()
+
+    mock_compute_parameter_update_loss.assert_called_once()
+    mock_increase_momentum_parameters.assert_not_called()
+    mock_accept_matrix_updates.assert_not_called()
+    mock_decrease_momentum_parameters.assert_called_once()
+    mock_reject_matrix_updates.assert_called_once()
+
+    mock_compute_parameter_update_loss.reset_mock()
+    mock_decrease_momentum_parameters.reset_mock()
+    mock_reject_matrix_updates.reset_mock()
+
+    # loss constant
+    mock_compute_parameter_update_loss.return_value = 5.0
+    kernel.step()
+    mock_compute_parameter_update_loss.assert_called_once()
+    mock_compute_parameter_update_loss.ass
```

Comment:
Missing some text here. (Test doesn't fail because this line is a no-op.)

I assume this is just extraneous and can be deleted?

Reply:
Exactly, I deleted the line.